pkg/feature/manifest.go

```diff
@@ -1,76 +1,101 @@
 package feature
 
 import (
+	"bytes"
+	"embed"
 	"fmt"
 	"html/template"
-	"os"
+	"io"
+	"io/fs"
 	"path/filepath"
 	"strings"
-
-	"github.com/pkg/errors"
 )
 
-const BaseOutputDir = "/tmp/odh-operator"
+//go:embed templates
+var embeddedFiles embed.FS
 
 type manifest struct {
 	name,
-	path string
+	path,
+	processedContent string
 	template,
-	patch,
-	processed bool
+	patch bool
 }
 
-func loadManifestsFrom(path string) ([]manifest, error) {
+func loadManifestsFrom(fsys fs.FS, path string) ([]manifest, error) {
 	var manifests []manifest
-	if err := filepath.Walk(path, func(path string, info os.FileInfo, err error) error {
+
+	err := fs.WalkDir(fsys, path, func(path string, dirEntry fs.DirEntry, walkErr error) error {
+		_, err := dirEntry.Info()
 		if err != nil {
 			return err
 		}
-		if info.IsDir() {
+
+		if dirEntry.IsDir() {
 			return nil
 		}
-		basePath := filepath.Base(path)
-		manifests = append(manifests, manifest{
-			name:     basePath,
-			path:     path,
-			patch:    strings.Contains(basePath, ".patch"),
-			template: filepath.Ext(path) == ".tmpl",
-		})
+		m := createManifestFrom(path)
+		manifests = append(manifests, m)
 
 		return nil
-	}); err != nil {
-		return nil, errors.WithStack(err)
+	})
+
+	if err != nil {
+		return nil, err
 	}
 
 	return manifests, nil
 }
 
+func createManifestFrom(path string) manifest {
+	basePath := filepath.Base(path)
+	m := manifest{
+		name:     basePath,
```

Comment:
Wouldn't it make sense to add a reference to the file system where the manifest is located? That way, `process` won't need the fsys argument, and it will be cleaner if we ever use several `fs.FS`.

Reply:
Although... well... the FS seems to be stored in the `Feature`.

Reply:
In fact, this is an aspect of manifest (the source of it), not a feature per se, so it should belong to the manifest struct instead. I took the liberty and reworked it in 588e61c /cc @cam-garrison

Reply:
Thanks for the great suggestion @israel-hdez